shared/src/main/scala/pl/metastack/metatime/Time.scala

```diff
@@ -18,13 +16,82 @@ trait Component extends Ordered[Component] {
   def unix(): Unix
 
   override def compare(that: Component): Int =
-    unix().value.compare(that.unix().value)
+    this.unix().value compare that.unix().value
+	
+  
+  val timeComponents = List ("Millisecond", "Second", "Minute", "Hour", "Time")
+  val dateComponents = List("Date", "DateTime")
+  
+  def stripClassName(name : String) : String = {
+    var temp = name.split('$')(0).split(' ')(1)
+    temp.substring(temp.lastIndexOf(".")+1)
+  }
+ 
+  def operationResultType(other: Component) : String = { 
+    val thisClass = stripClassName(this.getClass.toString())
+    val otherClass = stripClassName(other.getClass.toString())
+    if (thisClass.equals(otherClass))
+      timeComponents(4)
+    else if (dateComponents.contains(thisClass) || dateComponents.contains(otherClass))
+      dateComponents(1)
+    else if (timeComponents.contains(thisClass) && timeComponents.contains(otherClass))
+      timeComponents(4)
+    else
+      null
+  }
+  
+  def getTimeObject(comp : Component) : Time = {
```

Comment:
Is there a particular reason why you don't use pattern matching here? How about:

```scala
comp match {
  case hour: Hour => Time(hour = hour.h)
  ...
}
```

Also, we should not use `null`.